File: dotnet/Allmark/Inline/DeletionRule.cs
namespace Allmark.Inline;

using Allmark.Types;

public static class DeletionRule
{
    public static InlineRule Create()
    {
        return new InlineRule
        {
            Name = "deletion",
            Test = TestDeletion,
            Precedence = 20,
        };
    }

    private static bool TestDeletion(InlineParserState state, MarkdownNode parent)
    {
        var rule = Create();
        return CriticMarksRule.Execute("deletion", "-", state, parent, rule.Precedence!.Value);
    }
}

File: web/src/inline/deletionRule.ts
import type InlineParserState from "../types/InlineParserState";
import type InlineRule from "../types/InlineRule";
import type MarkdownNode from "../types/MarkdownNode";
import testCriticMarks from "./criticMarksRUle";

const rule: InlineRule = {
	name: "deletion",
	test: testDeletion,
};
export default rule;

function testDeletion(state: InlineParserState, parent: MarkdownNode): boolean {
	return testCriticMarks(rule.name, "-", state, parent);
}
